### PR TITLE
Enhance setting renaming

### PR DESCRIPTION
**Pull request by Hsu-Cheng-Wei**
Hello I modify this https://github.com/ErikEJ/EFCorePowerTools/pull/316
and I provider a better way to setting name by configure file, but this method can't not distinguish schema name

### CODE REVIEW

File: src/GUI/ReverseEngineer20/ReverseEngineer/ReplacingCandidateNamingService.cs [hunk 1]
```diff
@@ -36,8 +36,8 @@ public override string GenerateCandidateIdentifier(DatabaseTable originalTable)
 
             if (schema.UseSchemaName)
             {
-                candidateStringBuilder.Append(ToPascalCase(originalTable.Schema));
-                candidateStringBuilder.Append(ToPascalCase(originalTable.Name));
+                candidateStringBuilder.Append(originalTable.Schema.ToPascalCase());
+                candidateStringBuilder.Append(originalTable.Name.ToPascalCase());
```
Comment (ErikEJ) [inline src/GUI/ReverseEngineer20/ReverseEngineer/ReplacingCandidateNamingService.cs]: Where is ToPascalCase defined now?

**Comment by ErikEJ** [comment]
Could you add some tests?

**Comment by Hsu-Cheng-Wei** [comment]
Sure I push unit test for it

File: src/GUI/UnitTests/PluralizerTest.cs [hunk 2]
```diff
@@ -405,5 +407,68 @@ public void Issue221(string word, string expectedResult)
...
+                  }
+              };
+
+            var sut = new InflectorPluralizer(exampleOption);
+
+            //Act
+            var actResult = sut.Pluralize("OldTable");
+
+            //Assert
+            StringAssert.Contains(expected, actResult);
```
Comment (ErikEJ) [inline src/GUI/UnitTests/PluralizerTest.cs]: Why Contains?

File: src/GUI/UnitTests/PluralizerTest.cs [hunk 3]
```diff
@@ -405,5 +407,68 @@ public void Issue221(string word, string expectedResult)
...
+                  new Schema
+                  {
+                       SchemaName = "machine",
+                       UseSchemaName = false,
+                       Tables = new List<TableRenamer>
+                       {
+                          new TableRenamer
+                          {
+                            Name = "old_table",
+                            VariableName = "TableVariables",
```
Comment (ErikEJ) [inline src/GUI/UnitTests/PluralizerTest.cs]: Could we find a better name?

File: src/GUI/UnitTests/PluralizerTest.cs [hunk 4]
```diff
@@ -405,5 +407,68 @@ public void Issue221(string word, string expectedResult)
             // Assert
             Assert.That(result, Is.EqualTo(expectedResult));
         }
+
+        [Test]
+        public void InflectorPluralizerVariables()
```
Comment (ErikEJ) [inline src/GUI/UnitTests/PluralizerTest.cs]: Could you add a test that shows that the original issue is fixed?

File: src/GUI/ReverseEngineer20/ReverseEngineer/InflectorPluralizer.cs [hunk 5]
```diff
@@ -1,14 +1,36 @@
 ﻿using Humanizer;
 using Microsoft.EntityFrameworkCore.Design;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace ReverseEngineer20.ReverseEngineer
 {
     public class InflectorPluralizer : IPluralizer
     {
+        private readonly Dictionary<string, string> _customerCharacterPluralize;
```
Comment (ErikEJ) [inline src/GUI/ReverseEngineer20/ReverseEngineer/InflectorPluralizer.cs]: _customer => _custom

File: src/GUI/ReverseEngineer20/ReverseEngineer/InflectorPluralizer.cs [hunk 6]
```diff
@@ -1,14 +1,36 @@
...
+        private readonly Dictionary<string, string> _customerCharacterSingularize;
+
+        public InflectorPluralizer(List<Schema> customReplacers)
+        {
+            _customerCharacterPluralize = customReplacers
+                .SelectMany(x => x.Tables)
+                .Where(x => !string.IsNullOrWhiteSpace(x.VariableName))
+                .ToDictionary(key => key.Name.ToPascalCase(), value => value.VariableName);
+
+            _customerCharacterSingularize = customReplacers
```
Comment (ErikEJ) [inline src/GUI/ReverseEngineer20/ReverseEngineer/InflectorPluralizer.cs]: Will this not always block standard renaming from working?

Comment (Hsu-Cheng-Wei) [reply src/GUI/ReverseEngineer20/ReverseEngineer/InflectorPluralizer.cs]: I don't known what's your meaning
I think If you want to custom name, you only want your name is equals your naming not be transfer by other function?

**Comment by ErikEJ** [comment]
This looks good, will just let it sit for while until the EF Core 3.1 dust settles

**Comment by ErikEJ** [comment]
Closing in favor of: https://github.com/ErikEJ/EFCorePowerTools/pull/496

**Pull request closed without merging**